### PR TITLE
Tidy & fix deprecation warnings

### LINKED ISSUE #23031: Setting site to deprecated does not show deprecation warning in documentation section
**This is a Bug Report**

**Problem:**
If you make a new Kubernetes release and publish the branch up to that release as a deprecated website, the deprecated website for the just-superseded minor version **doesnt** show a deprecation warning for pages inside `/docs/`.

**Proposed Solution:**
Fix the layout to show a deprecation warning as expected.

**Page to Update:**
https://kubernetes.io/docs/ and contentns

**Kubernetes Version:**
v1.18

**Additional Information:**
/kind bug
/priority important-soon

### PR DESCRIPTION
**Pull request by sftim**
Fixes #23031

To preview these changes, check out locally, eg:
```
git fetch upstream pull/22565/head:pr-22032
git checkout pr-22032
```
then set `deprecated = true` inside `config.toml` and preview, eg: `make container-serve`.

I also recommend previewing without an announcement visible; we don't put announcements on archived websites.

### CODE REVIEW

**Comment by zacharysarah** [comment]
Quick fix for the issue description:
```suggestion
git fetch upstream pull/22565/head:pr-23032
git checkout pr-23032
```

**Comment by zacharysarah** [comment]
Local tests show no deprecation banner.

I also had to add a couple of files to resolve build errors related to the impending Persian localization:
- `content/fa/_common-resources/index.md`
- `content/fa/case-studies/_index.md`

I added a simple front matter to both:
```
---
headless: true
---
```

Screen Shot 2020-08-12 at 11 58 41 AM

**Comment by celestehorgan** [comment]
When I pull the branch down, I intend to test this by: 

1) Change`config.toml`'s `deprecated` setting to `true`.
2) Build using `make container-serve`
3) Ensure that the deprecation warning shows up on all pages that use the different layouts covered in the PR.

If at all possible, I think the most thorough way to test this would be to (if possible) test out this PR with whatever docs snapshotting process takes place when we version.

**Comment by jimangel** [comment]
I think the two most important areas to test are:
- The main landing page says THESE ARE OLD DOCS
- Any sub-doc pages (pick a few) have the THESE ARE OLD DOCS warning

It might be worth checking a few other areas, but those two are the ones we really should focus on incase somebody lands on an old doc from a google search or toggles versions and forgets.

**Comment by celestehorgan** [comment]
Based on the above comments, and run from the container image:

The deprecation warning..

- Does load on `/_index.html` (the site homepage)
- Does not load on `/docs`
- Does not load on individual docs pages, i.e. `/docs/setup`
- Does not load on `/blog`
- Does not load on individual blog pages, i.e. `/blog/2020/08/03/physics-politics-and-pull-requests-the-kubernetes-1.18-release-interview/`
- Does not load on `/case-studies`
- Does not load on individual case study pages, i.e. `/case-studies/denso`

>** Note:** when I say "does not load", I mean "Does not load _or_ is not visible, i.e covered by another element"


---

@sftim: I would like someone else to confirm these results before you spend time going in to fix them. @onlydole @jimangel – could one of you pull down this branch, run using the instructions above and confirm/deny?

**Comment by kbhawkey** [comment]
Looking now.

File: layouts/partials/deprecation-warning.html [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{ if .Site.Param "deprecated" }}
+{{ if (.Site.Param "deprecated") }}
```
Comment (kbhawkey) [inline layouts/partials/deprecation-warning.html:1]: Is this change needed?

**Comment by kbhawkey** [comment]
Hi @sftim 
I tried both branches -- went with the second 23032 and removed the `fa` directory.
The changes were not pulled down.
Anyway, I added the  `content-page`, `docs/baseof`, and `deprecation-warning` template modifications.
The warning appears on the docs pages. I'll add in the new styling too.
Can you provide details as to what changed (theme updates)?
I'd like to know what worked in the past.

**Comment by kbhawkey** [comment]
Here is more feedback:
The deprecation warning has a yellow background and does not fill in the width of the page. To me, it looks a bit like
a sticky note applied to a web page. 
I have not looked at the css for the partial. May want to check if the left margin is the same as the content margin.
The warning appears on docs and blog pages in addition to case studies pages. The deprecation warning only appears on the case studies home page, `/case-studies/`.
On my screen, the "edit" or "pencil" icon overlaps with the deprecated div.
The warning appears on section-index pages (the edit button does not).
Question: Do you want the deprecation warning to display as a banner in the header instead of in specific pages?

**Comment by sftim** [comment]
Overall, my aim for this PR is to have at least minimum-viable deprecation warnings in place before the v1.19 release.
I'm worried that the perfect may be the enemy of the good enough to merge.

I'm very happy to see iterative improvements beyond that.
If any of these changes introduce regressions or require amendments before a merge, I can pick up those fixes.

**Comment by kbhawkey** [comment]
> 
> 
> Overall, my aim for this PR is to have at least minimum-viable deprecation warnings in place before the v1.19 release.
> I'm worried that the perfect may be the enemy of the good enough to merge.
> 
> I'm very happy to see iterative improvements beyond that.
> If any of these changes introduce regressions or require amendments before a merge, I can pick up those fixes.

Hi @sftim. Thanks for working on fixing the deprecation warnings for the release.
My comments are intended to help not push for "perfection". Getting this fixed is very important.
I tried a few experiments, locally. What do you think about changing three baseof files (instead of the content-page), config.toml, and the style updates. Afterwards, the .css file could be removed.

I updated the header sections in the baseof.html files, adding the deprecated-warning partial into the header.
I added a check for the Case studies in the _default baseof. I did not fully test these changes, but the warning displays.

```
layouts/blog/baseof.html
<header>
      {{ partial "navbar.html" . }}
      {{ partial "announcement.html" . }}
      <section class="header-hero text-center text-white font-bold pb-4">
        <h1>
          {{ $sectionHeading := .Site.GetPage "section" .Section }}
          {{ with $sectionHeading }}
            {{ .Title }}
          {{ end }}
        </h1>
      </section>
      {{ block "deprecated" . }}
      {{ partial "deprecation-warning.html" . }}
      {{ end }}
    </header>
```
```
layouts/docs/baseof.html
<header>
      {{ partial "navbar.html" . }}
      {{ partial "announcement.html" . }}
      <section class="header-hero text-center text-white font-bold pb-4">
        <h1>
          {{ $sectionHeading := .Site.GetPage "section" .Section }}
          {{ with $sectionHeading }}
            {{ .Title }}
          {{ end }}
        </h1>
      </section>
      {{ block "deprecated" . }}
      {{ partial "deprecation-warning.html" . }}
      {{ end }}
    </header>
```

```
layouts/_defaults/baseof.html
{{ block "hero" . }}
		<!--  HERO  -->
		<section class="header-hero text-white pb-4 light-text">
			<div class="main-section">
				<div id="headlineWrapper">
					<h1>{{ block "hero-title" . }}{{ .Params.bigheader | default .Title }}{{ end }}</h1>
					<h5>{{ .Params.abstract }}</h5>
				</div>
				{{ block "hero-more" . }}{{ end }}
			</div>
		</section>
		{{ block "post-hero" . }}
      {{ block "deprecated" . }}
        {{ if .IsHome }}
          {{ partial "deprecation-warning.html" . }}
        {{ else if eq .Params.cid "caseStudies" }}
        {{ partial "deprecation-warning.html" . }}
        {{ end }}
      {{ end }}
    {{ end }}
		{{ end }}
```

**Comment by kbhawkey** [comment]
Here are two pics from testing:
deprecate-warn-blog
deprecate-warn-doc

**Comment by sftim** [comment]
Putting the deprecation warning in the header detaches the sidebar border from the box it's supposed to meet at the top, so I'm not keen on that approach.

How about this? The CSS is a bit nasty - happy to tidy that up (might have to get some Sass practice in first).

**Comment by celestehorgan** [comment]
@sftim I can jump in on the SCSS if you like :)

**Comment by sftim** [comment]
Is this OK to merge now and tidy later?
